app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedHighwayConstruction.java

```diff
@@ -0,0 +1,119 @@
+package de.westnordost.streetcomplete.quests.construction;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+import android.util.Log;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import de.westnordost.osmapi.map.data.BoundingBox;
+import de.westnordost.streetcomplete.R;
+import de.westnordost.streetcomplete.data.meta.OsmTaggings;
+import de.westnordost.streetcomplete.data.osm.changes.StringMapChangesBuilder;
+import de.westnordost.streetcomplete.data.osm.download.MapDataWithGeometryHandler;
+import de.westnordost.streetcomplete.data.osm.download.OverpassMapDataDao;
+import de.westnordost.streetcomplete.data.osm.tql.OverpassQLUtil;
+import de.westnordost.streetcomplete.quests.DateHandler;
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedHighwayConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedHighwayConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with invalid construction tag
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * - with fixme (as in some situations this quest may add fixme note as an edit)
+	 * . */
+	private static String getOverpassQuery(BoundingBox bbox)
+	{
+		ArrayList<String> acceptedConstructionValues = new ArrayList<String>(Arrays.asList(OsmTaggings.ALL_ROADS));
+		acceptedConstructionValues.add("cycleway");
+		acceptedConstructionValues.add("footway");
+		acceptedConstructionValues.add("path");
+
+		String groupName = ".roads_under_construction";
+		String resultsName = ".roads_for_review";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			getQueryPart("highway", groupName, 14) +
+			"(" +
+			"way[construction]" +
```

Comment:
Isn't this line superfluous?

Reply:
Also, why are "invalid road construction types" filtered out?

Reply:
How about not excluding it, but treating these kind of invalid values in a way that then the quest tags the fallback `highway=road`? I mean, in the wiki, it is pretty clearly defined that the value of the construction tag should be the value of the highway tag, right? So, if it is invalid, then it can be ignored. But what shouldn't be ignored is the fact that the construction site exists at all.

Where does `service_parking` come from? Was that just some guy or is this used by several people?

Reply:
Or, another option would be to tag `highway=service_parking` in that case. Why should the app fix tagging anomalies automatically that were introduced by humans?

Reply:
OK, so such edit would not introduce a mistake - it may even make it more visible.

And useful data (that construction is completed) is still added.

And it will simplify code.

I think that keeping it makes no sense (it remains from original version of code).